analyses/tumor-purity-exploration/01_explore-tumor-purity.Rmd

```diff
@@ -0,0 +1,503 @@
+```
+
+Distribution by cancer group and broad histology:
+
+```{r}
+# combine metadata with palette:
+tumor_palette_df <- metadata_tumor_wgs %>%
+  inner_join(palette_mapping_df) %>%
+  select(Kids_First_Biospecimen_ID, 
+         tumor_fraction, 
+         cancer_group_display, 
+         cancer_group_hex, 
+         broad_histology_display,
+         broad_histology_order,
+         broad_histology_hex) %>%
+  drop_na() %>%
+  # reorder by median fraction and ensure "Other" is always last
+  mutate(cancer_group_display = forcats::fct_reorder(cancer_group_display, tumor_fraction), 
+         cancer_group_display = forcats::fct_relevel(cancer_group_display, "Other", after = Inf),
+         broad_histology_display = forcats::fct_reorder(broad_histology_display, tumor_fraction), 
+         broad_histology_display = forcats::fct_relevel(broad_histology_display, "Other", after = Inf))
+         
+# plotting function:
+tumor_violins <- function(df, display_name, display_hex) {
+  ggplot(df) + 
+    # order by ascending median
+    aes(x = forcats::fct_reorder({{display_name}}, tumor_fraction),
+        y = tumor_fraction, 
+        fill = {{display_hex}}) + 
+    geom_violin(alpha = 0.5) +
+    stat_summary() + 
+    scale_fill_identity() + 
+    theme(axis.text.x = element_text(angle = 30, hjust = 1)) 
+}
+
+# Cancer group
+tumor_violins(tumor_palette_df, cancer_group_display, cancer_group_hex)
+
+# Broad histology
+tumor_violins(tumor_palette_df, broad_histology_display, broad_histology_hex)
+```
+
+
+
+
+Distribution across sequencing centers, highlighting cancer groups:
+
+```{r}
+# use jitter to highlight diagnoses
+seq_center_df <- tumor_palette_df %>%
+  inner_join(
+    select(metadata, 
+           Kids_First_Biospecimen_ID, 
+           seq_center)
+  )
+
+ggplot(seq_center_df) +
+  aes(x = seq_center, 
+      y = tumor_fraction, 
+      color = cancer_group_hex) + 
+  geom_jitter() + 
+  scale_color_identity() +
+  theme(axis.text.x = element_text(angle = 30, hjust=1))
+
+# What are the cancer groups at the smaller centers?
+seq_center_df %>%
+  count(seq_center, cancer_group_display) %>%
+  arrange(seq_center)
+```
+
+
+## Tumor fraction at the RNA level
+
+Tumor fractions explored above were derived from WGS data.
+Estimating tumor fraction from RNA directly is challenging since methods focus on estimating proportion of immune cells and assuming the rest is tumor, which is not a fair assumption for many of these diagnoses.
+
+Therefore, for RNA data tumor purity estimates, we should instead directly use tumor fraction values but _only for those samples where DNA and RNA sequencing were performed on the same extraction_.
+We cannot really know the RNA tumor purity for samples that have separate DNA/RNA extractions.
+
+This extraction information is present in data release `>=v23`.
+Only RNA are annotated in this column, hence the `NA`s below:
+
+```{r}
+table(metadata$extraction_type, useNA = "ifany")
```

Comment:
I think this is similar to my comment about the presence table above – if the `NAs` aren't useful, maybe we should drop them before you make the table.